parser/openfoodfacts_taxonomy_parser/parser/taxonomy_parser.py

```diff
@@ -438,6 +444,54 @@ def _get_valid_child_links(self, entry_nodes: list[NodeData], raw_child_links: l
 
         return valid_child_links
 
+    def _remove_duplicate_child_links(self, child_links: list[ChildLink]) -> list[ChildLink]:
+        """Remove duplicate child links (i.e child links with the same parent_id and id)"""
+        unique_child_links = []
+        children_to_parents = collections.defaultdict(set)
+        for child_link in child_links:
+            parent_id, child_id = child_link["parent_id"], child_link["id"]
+            if parent_id not in children_to_parents[child_id]:
+                children_to_parents[child_id].add(parent_id)
+                unique_child_links.append(child_link)
+        return unique_child_links
+
+    def _merge_duplicate_entry_nodes(self, entry_nodes: list[NodeData]) -> list[NodeData]:
+        """Merge entry nodes with the same id:
+        - merge their tags (union)
+        - merge their properties (union, and in case of conflict, keep the last value)"""
+        unique_entry_nodes = []
+        ids_to_nodes = dict()
+        for node in entry_nodes:
+            if node.id in ids_to_nodes:
+                first_node = ids_to_nodes[node.id]
+                for key, value in node.tags.items():
+                    if not key.startswith("tags_ids_"):
+                        # union of the tags
+                        first_node.tags[key] = list(
+                            dict.fromkeys(first_node.tags.get(key, []) + value)
+                        )
```

Comment:
why not a set ?
```suggestion
                        first_node.tags[key] = list(
                            set(first_node.tags.get(key, []) + value)
                        )
```

Reply:
because we lose the order of the translations if we convert to a set!

Reply:
Ok, then add a comment to explain you use it to keep ordering.